### PR TITLE
feat(api): keep sign-in sessions active for eight days

### PR DESCRIPTION
**Pull request by ajfisher**
## Behavioural claim

New magic-link sign-ins remain authenticated for an absolute eight-day lifetime, while sessions issued before this change keep their original stored expiry.

## Specification and acceptance evidence

| Acceptance criterion | Evidence | Result |
| --- | --- | --- |
| Local server and Lambda defaults use one eight-day value; deployed configuration is 691,200 seconds | `api/src/tests/session.test.ts`; `api/src/tests/deployment-config.test.ts`; `serverless.api-core.yml` | PASS |
| Magic-link completion aligns response expiry, absolute cookie expiry, DynamoDB logical expiry, and TTL cleanup epoch | `api/src/tests/magic-link.test.ts`; `api/src/tests/lambda-core.test.ts` | PASS |
| A session is valid through expiry minus one second and invalid at the exact expiry instant | `api/src/tests/magic-link.test.ts` fixed-clock boundary case | PASS |
| Fractional-second issuance remains active for a full eight days and the browser cookie cannot outlive the persisted server boundary | Fractional-clock persistence test plus cookie/Lambda assertions for absolute `Expires` and absence of response-relative `Max-Age` | PASS |
| Existing 24-hour sessions remain usable until their stored boundary and are not extended by the new configuration | `api/src/tests/magic-link.test.ts` legacy/new-session transition case | PASS |
| Token TTL, auth route protection, cookie flags, CSP, and security headers remain intact | Full repository test suite: 273 API, 93 app, and 57 review-gate tests | PASS |
| UX, engineering, and quality independent reviews find no material implementation or evidence issue | Read-only local agent reviews, including re-audit of the absolute-expiry fix | PASS |
| The exact PR head deploys to QA and core/auth endpoints pass smoke checks | `deploy-qa` run 31937859381 for `5dff29376768aee368af86bb742ff2b5ec7af517` | PASS |
| Codex independent review covers session boundary, rollback, and INV-009 | Exact-head Codex review comment for `5dff29376768aee368af86bb742ff2b5ec7af517`: no major issues | PASS |

## Scope boundaries

Included:

- Eight-day absolute session lifetime for newly completed magic-link sign-ins.
- Shared local/Lambda default, QA/production Serverless configuration, strict logical expiry, tests, and specification documentation.

Excluded:

- Magic-link token lifetime, which remains 15 minutes.
- Sliding renewal, migration or extension of existing sessions, logout UI, bulk revocation, API response shape, and DynamoDB schema changes.

## Change classification

- Declared risk: `high`
- [ ] `documentation-only` — Documentation only
- [ ] `tests-only` — Tests only
- [x] `application-behaviour` — Application or API behaviour
- [ ] `backlog-maintenance` — Canonical backlog maintenance
- [ ] `dependency-tooling` — Dependency or tooling
- [ ] `public-contract` — Public API or repository contract
- [ ] `data-migration` — Database schema or data migration
- [ ] `permission-trust-boundary` — Permission or trust boundary
- [ ] `durable-state-ownership` — Durable state or ownership
- [ ] `destructive-behaviour` — Destructive or difficult-to-reverse behaviour
- [ ] `new-production-dependency` — New production dependency
- [x] `authentication-authorisation` — Authentication or authorisation
- [ ] `privacy-regulated-data` — Privacy or regulated data
- [x] `infrastructure-production-configuration` — Infrastructure or deployment control
- [ ] `review-policy` — Review policy, packet, or workflow

## Architecture and invariants

- [ ] `architecture:none`
- [x] `architecture:documented`
- [ ] `architecture:judgement-required`

### Affected invariants

| Invariant | How this PR affects it | Why it remains valid | Evidence |
| --- | --- | --- | --- |
| INV-009 | Extends the absolute lifetime of session cookies and their matching server-side records from one day to eight days. | Cookies remain HttpOnly, SameSite=Lax, environment-appropriate Secure, host-only, and path-scoped; logical expiry remains authoritative even when DynamoDB TTL cleanup is delayed. | Lambda cookie integration test, fixed-clock logical-expiry tests, full app security-header suite, and pending deployed QA configuration inspection. |

### Architecture or decision record

`docs/spec.md` section 10 now records the eight-day, absolute, non-sliding lifetime and the decision not to rewrite existing session expiries. No storage ownership, schema, or public response shape changes.

### Failure behaviour

The longer bearer-cookie lifetime increases exposure on a lost or shared device because the current product has no user-facing logout or bulk revocation workflow. A stolen cookie may remain usable for up to eight days. Existing HttpOnly, Secure, SameSite, origin, CSP, and server-side logical-expiry protections remain unchanged. Deployment drift is detected by an exact Serverless configuration test; logical expiry rejects at the stored expiry even if DynamoDB cleanup is late.

### Rollback approach

Revert this PR to restore a 24-hour lifetime for newly issued sessions. That configuration rollback does not shorten sessions already issued with an eight-day stored expiry. Emergency revocation requires deleting the specifically identified `AUTH_SESSION#<sessionId>` row; a broad session purge is an explicit operator decision and is not automated by this PR.

### Rollback evidence

The fixed-clock legacy-session regression proves configured lifetime changes do not rewrite stored expiries. The deployment-config test pins the environment value, and `npm test` verifies strict logical expiry plus invalid-session handling. No data migration is required for either forward deployment or code rollback.

### Unresolved blocking findings

None.

### Rejected findings and evidence

None.

## Human judgement

- [x] `human-judgement:none`
- [ ] `human-judgement:required`

### Decision requiring judgement

None.

### Options considered

None.

### Reason selected

None.

### Reversal cost

None.

## Review focus

Review the exact-expiry boundary, alignment of cookie and persisted expiry, preservation of pre-existing session lifetimes, deployment-value drift, INV-009 protections, and the non-immediate rollback consequence for already-issued eight-day sessions.

### CODE REVIEW

**Comment by ajfisher** [comment]
@codex review for the exact current head ea89dd0e66e825a95adc5cb347b1c3434e683716, focusing on INV-009, exact session-expiry boundaries, cookie/persisted-expiry alignment, existing-session continuity, deployment configuration, failure behaviour, and rollback safety.

File: api/src/auth/magic-link.ts [hunk 1]
```diff
@@ -384,7 +384,7 @@ export class MagicLinkService {
     const nowEpoch = Math.floor(this.clock.now().getTime() / 1000);
     const expiresAtEpoch = readNumber(item.expiresAtEpoch, "expiresAtEpoch");
 
-    if (expiresAtEpoch < nowEpoch) {
+    if (expiresAtEpoch <= nowEpoch) {
```
[the remark this answers is not in the document]
Comment (ajfisher) [reply api/src/auth/magic-link.ts:387]: Addressed in 5dff293. Both local and Lambda handlers now construct the cookie from the persisted absolute expiresAt value using Expires, with no response-relative Max-Age to override it. Session issuance rounds fractional timestamps up to the next epoch second so the stored lifetime is at least eight full days, and the new fractional-clock regression proves validity through the shared boundary. Full typecheck, 275 API tests, 93 app tests, 57 review-gate tests, and contract checks pass.

**Comment by ajfisher** [comment]
@codex review the exact current head 5dff29376768aee368af86bb742ff2b5ec7af517 after the boundary fix. Focus on absolute cookie/persisted-expiry alignment, fractional-second lifetime, INV-009, existing-session continuity, deployment configuration, failure behaviour, and rollback safety.
